src/app/reporting/Engine.cpp

```diff
@@ -645,14 +643,14 @@ void Engine::Run()
 
     // We may be deallocating read handlers as we go.  Track how many we had
     // initially, so we make sure to go through all of them.
-    size_t initialAllocated = mpImEngine->mReadHandlers.Allocated();
+    size_t initialAllocated = mDelegate->GetReadHandlerPool().Allocated();
     while ((mNumReportsInFlight < CHIP_IM_MAX_REPORTS_IN_FLIGHT) && (numReadHandled < initialAllocated))
     {
         ReadHandler * readHandler =
-            mpImEngine->ActiveHandlerAt(mCurReadHandlerIdx % (uint32_t) mpImEngine->mReadHandlers.Allocated());
+            mDelegate->ActiveHandlerAt(mCurReadHandlerIdx % (uint32_t) mDelegate->GetReadHandlerPool().Allocated());
```

Comment:
Either delegate should have a way to get the count, or delegate should have a method for whatever the actual operation here is.